code/game/machinery/computer/arcade.dm

```diff
@@ -1075,3 +1075,272 @@
 #undef ORION_STATUS_NORMAL
 #undef ORION_STATUS_GAMEOVER
 #undef ORION_STATUS_MARKET
+
+//////////////////
+// Claw Machine //
+//////////////////
+
+/obj/machinery/computer/arcade/clawmachine
+	name = "AlliCo Grab-a-Gift"
+	desc = "Show off your arcade skills for that special someone!"
+	icon_state = "clawmachine"
+	icon_keyboard = null
+	icon_screen = null
+	circuit = /obj/item/weapon/circuitboard/arcade/clawmachine
+	prizes = list(/obj/random/plushie)
+	var/wintick = 0
+	var/winprob = 0
+	var/instructions = "Insert 1 thaler or swipe a card to play!"
+	var/gameStatus = "CLAWMACHINE_NEW"
+	var/gamepaid = 0
+	var/gameprice = 1
+	var/winscreen = ""
+
+/// Payment
+/obj/machinery/computer/arcade/clawmachine/attackby(obj/item/I as obj, mob/user as mob)
+	if(..())
+		return
+
+	var/obj/item/weapon/card/id/W = I.GetID()
+
+	if(!emagged)
+		if(gamepaid == 0 && vendor_account && !vendor_account.suspended)
+			var/paid = 0
+			var/handled = 0
+
+			if(W) //for IDs and PDAs and wallets with IDs
+				paid = pay_with_card(W,I)
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash/ewallet))
+				var/obj/item/weapon/spacecash/ewallet/C = I
+				paid = pay_with_ewallet(C)
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash))
+				var/obj/item/weapon/spacecash/C = I
+				paid = pay_with_cash(C, user)
+				handled = 1
+
+			if(paid)
+				gamepaid = 1
+				return
+			else if(handled)
+				return // don't smack that machine with your 2 thalers
+	else if(emagged)
+		if(gamepaid == 0 && vendor_account && !vendor_account.suspended)
+			var/paid = 0
+			var/handled = 0
+
+			if(W) //for IDs and PDAs and wallets with IDs
+				paid = pay_with_card(W,I)
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash/ewallet))
+				playsound(src, 'sound/arcade/steal.ogg', 50, 1, extrarange = -3, falloff = 0.1, ignore_walls = FALSE)
+				to_chat(user, "<span class='info'>It doesn't seem to accept that! Seem you'll need to swipe a valid ID.</span>")
+				handled = 1
+			else if(istype(I, /obj/item/weapon/spacecash))
+				playsound(src, 'sound/arcade/steal.ogg', 50, 1, extrarange = -3, falloff = 0.1, ignore_walls = FALSE)
+				to_chat(user, "<span class='info'>It doesn't seem to accept that! Seem you'll need to swipe a valid ID.</span>")
+				handled = 1
+			if(paid)
+				gamepaid = 1
+				instructions = "Hit start to play!"
+				return
+			else if(handled)
+				return // don't smack that machine with your 2 thalers

```

Comment:
Handled doesn't do anything here, because the code would naturally fall out the end of this block and hit the end of the proc anyways.